process/block/baseProcess.go

```diff
@@ -801,6 +793,50 @@ func (bp *baseProcessor) getLastCrossNotarizedHeadersForShard(shardID uint32) *b
 	return headerInfo
 }
 
+func (bp *baseProcessor) getLastSelfNotarizedHeaders() []bootstrapStorage.BootstrapHeaderInfo {
+	lastSelfNotarizedHeaders := make([]bootstrapStorage.BootstrapHeaderInfo, 0)
```

Comment:
it would be faster to prealocate numOfShardsElements and use index instead of append.

Reply:
Done